packages/next/src/server/lib/incremental-cache/fetch-cache.ts

```diff
@@ -168,8 +168,15 @@ export default class FetchCache implements CacheHandler {
     // on successive requests
     let data = memoryCache?.get(key)
 
-    // get data from fetch cache
-    if (!data && this.cacheEndpoint) {
+    const isCacheable = !data && this.cacheEndpoint
+    const hasFetchKindAndIncludesTags =
+      tags?.every(
+        (tag) => data?.value?.kind === 'FETCH' && data.value.tags?.includes(tag)
+      ) ?? false
+
+    // Get data from fetch cache. Also check if new tags have been
+    // specified with the same cache key (fetch URL)
+    if (isCacheable || hasFetchKindAndIncludesTags) {
```

Comment:
```suggestion
    if (isCacheable || !hasFetchKindAndIncludesTags) {
```
This should request data cache when not includes the tags right?